rllib/agents/dqn/simple_q.py

```diff
@@ -41,113 +43,203 @@
+    Example:
+        >>> config = SimpleQConfig()
+        >>> print(config.exploration_config)
+        >>> explore_config = config.exploration_config.update(
+        >>>     {
+        >>>         "type": "softq",
+        >>>         "temperature": [1.0],
+        >>>     }
+        >>> config = SimpleQConfig().training(lr_schedule=[[1, 1e-3], [500, 5e-3]])\
+        >>>                         .exploration(exploration_config=explore_config)
+    """
+
+    def __init__(self):
+        """Initializes a SimpleQConfig instance."""
+        super().__init__(trainer_class=SimpleQTrainer)
+
+        # Simple Q specific
+        # fmt: off
+        # __sphinx_doc_begin__
+        #
+        self.timesteps_per_iteration = 1000
+        self.target_network_update_freq = 500
+        self.replay_buffer_config = {
+            "_enable_replay_buffer_api": True,
+            "learning_starts": 1000,
+            "type": "MultiAgentReplayBuffer",
+            "capacity": 50000,
+            "replay_batch_size": 32,
+            "replay_sequence_length": 1,
+        }
+        self.store_buffer_in_checkpoints = False
+        self.lr_schedule = None
+        self.adam_epsilon = 1e-8
+        self.grad_clip = 40
+        self.learning_starts = 1000
+        # __sphinx_doc_end__
+        # fmt: on
+
+        # Overrides of TrainerConfig defaults
+        # `rollouts()`
+        self.num_rollout_workers = 0
+        self.rollout_fragment_length = 4
+
+        # `training()`
+        self.lr = 5e-4
+        self.train_batch_size = 32
+
+        # `exploration()`
+        self.exploration_config = {
+            "type": "EpsilonGreedy",
+            "initial_epsilon": 1.0,
+            "final_epsilon": 0.02,
+            "epsilon_timesteps": 10000,
+        }
+
+        # `evaluation()`
+        self.evaluation_config = {"explore": False}
+
+        # `reporting()`
+        self.min_time_s_per_reporting = 1
+
+        # `experimental()`
+        self._disable_execution_plan_api = True
+
+    @override(TrainerConfig)
+    def training(
+        self,
+        *,
+        timesteps_per_iteration: Optional[int] = None,
+        target_network_update_freq: Optional[int] = None,
+        replay_buffer_config: Optional[dict] = None,
+        store_buffer_in_checkpoints: Optional[bool] = None,
+        lr_schedule: Optional[List[List[Union[int, float]]]] = None,
+        adam_epsilon: Optional[float] = None,
+        grad_clip: Optional[int] = None,
+        learning_starts: Optional[int] = None,
+        **kwargs,
+    ) -> "SimpleQConfig":
+        """Sets the training related configuration.
+        Args:
+            timesteps_per_iteration: Minimum env steps to optimize for per train call. This value does not affect learning, only the length of iterations.
+            target_network_update_freq: Update the target network every `target_network_update_freq` steps.
+            replay_buffer_config: Replay buffer config.
+                Examples:
```

Comment:
Super nice! Thanks for adding these exaples. @ArturNiederfahrenhorst , keep these in mind when writing the documentation for the new replay buffer API/configs.